modules/packages/ArgumentParser.chpl

```diff
@@ -977,8 +977,11 @@ module ArgumentParser {
       _exitAfterHelp = exitAfterHelp;
 
       var _helpHandler = new shared HelpHandler();
+      // TODO: this fails if helpHandler is `borrowed`
       if !isNothingType(h) then
-        _helpHandler = helpHandler: shared h.chpl_t;
+        if isOwnedClass(helpHandler) || isUnmanagedClass(helpHandler)
+          then _helpHandler = shared.adopt(helpHandler);
```

Comment:
Do we still need to cast the `helpHandler` to `h.chpl_t`, besides the memory management change part?

Reply:
It is only needed when `h` is not `shared`, which the new code handles with `adopt`.